app/models/location.rb

```diff
@@ -121,6 +121,19 @@ class Location < ApplicationRecord
 
   scope :visible, -> { where(discarded_at: nil) }
 
+  scope :duplicate_sets, lambda {
+    scope = visible
+    .group(*DUPLICATE_LOCATION_ATTRIBUTES)
+    .where.not(scheme_id: nil)
+    .where.not(postcode: nil)
+    .where.not(mobility_type: nil)
+    .having(
+      "COUNT(*) > 1",
+    )
+    scope.pluck("ARRAY_AGG(id)")
+  }
+
+  DUPLICATE_LOCATION_ATTRIBUTES = %w[scheme_id postcode mobility_type].freeze
```

Comment:
This assumes that a location can only be a duplicate within the same scheme

Reply:
If there are duplicate schemes, presumably a location in one could be a duplicate of a location in the other?

Reply:
Yes! I thought that, but then if a user merges duplicate schemes (which at the moment they'd have to recreate all the locations from 2 duplicate schemes into 1) then those duplicate locations would get flagged.
ACs have this, which I wouldn't expect, cause it suggests we only compare locations of duplicate schemes. I'll double check that!
`Only compare the locations of duplicate schemes to other locations of the same scheme.`